### PR TITLE
Handle errors when parsing

### PR DESCRIPTION
**Pull request by mgax**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -36,6 +36,63 @@ pub use types::geometrycollection::GeometryCollection;
...
+    pub desc: &'static str,
+}
+
+impl GeoJsonError {
+    pub fn new(desc: &'static str) -> GeoJsonError {
+        return GeoJsonError{desc: desc};
+    }
+}
+
+pub type GeoJsonResult<T> = Result<T, GeoJsonError>;
```
Comment (frewsxcv) [inline src/lib.rs]: Why not `pub type GeoJsonResult<T> = Result<T, &str>;`?

Comment (mgax) [inline src/lib.rs]: I'm following the example of `IoError`, returning an object that can encode more information than just a message, maybe the line and column number in the original JSON document, if that's available.

**Comment by frewsxcv** [comment]
Is this ready to be reviewed/merged or do you have more to add?

**Comment by mgax** [comment]
It's ready.

File: src/lib.rs [hunk 2]
```diff
@@ -36,6 +36,63 @@ pub use types::geometrycollection::GeometryCollection;
 
 pub use types::feature::Feature;
 pub use types::featurecollection::FeatureCollection;
-pub use types::featurecollection::from_str;
+
+macro_rules! expect_string {
+    ($value:expr) => (try!(
+        match $value.as_string() {
```
Comment (frewsxcv) [inline src/lib.rs]: Depending on whether or not you want to use unstable features, you could utilize [`ok_or`](http://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or) here

Comment (mgax) [inline src/lib.rs]: For some reason, that didn't work, but I can't remember the details.

**Comment by frewsxcv** [comment]
Looks great! :sparkles:

**Comment by mgax** [comment]
Awesome, thanks!

**Pull request merged**